src/NavBar.ts

```diff
@@ -1,35 +1,40 @@
 import QueryStore from '@/stores/QueryStore'
 import * as QueryUrl from '@/QueryUrl'
+import MapOptionsStore from './stores/MapOptionsStore'
 
 export interface AppContext {
     addEventListener(type: string, listener: () => void): void
     readonly location: Location
     readonly history: History
 }
 export default class NavBar {
-    private readonly queryStore: QueryStore
     private readonly appContext: AppContext
+    private readonly queryStore: QueryStore
+    private readonly mapStore: MapOptionsStore
     private isIgnoreQueryStoreUpdates = false
 
-    constructor(queryStore: QueryStore, appContext: AppContext) {
-        this.queryStore = queryStore
-        this.queryStore.register(() => this.onQueryStoreChanged())
+    constructor(appContext: AppContext, queryStore: QueryStore, mapStore: MapOptionsStore) {
         this.appContext = appContext
         appContext.addEventListener('popstate', () => this.parseUrl())
+        this.queryStore = queryStore
+        this.queryStore.register(() => this.onQueryStateChanged())
+        this.mapStore = mapStore
+        this.mapStore.register(() => this.onQueryStateChanged())
     }
 
     parseUrl() {
         this.isIgnoreQueryStoreUpdates = true
-        QueryUrl.parseUrl(this.appContext.location.href, this.queryStore.state)
+        QueryUrl.parseUrl(this.appContext.location.href, this.queryStore.state, this.mapStore.state)
         this.isIgnoreQueryStoreUpdates = false
     }
 
-    private onQueryStoreChanged() {
+    private onQueryStateChanged() {
         if (this.isIgnoreQueryStoreUpdates) return
 
         const newHref = QueryUrl.createUrl(
             this.appContext.location.origin + this.appContext.location.pathname,
-            this.queryStore.state
+            this.queryStore.state,
+            this.mapStore.state
```

Comment:
Not sure where the "root" information of the map layer name should be. I kept it in the MapOptionsStore.

Reply:
I think I now know what you mean and I think it belongs into the ```MapOptionsStore```